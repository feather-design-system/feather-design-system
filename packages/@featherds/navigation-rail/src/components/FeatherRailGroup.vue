<template>
  <nav class="nav-rail-group" :aria-label="label">
    <FeatherList>
      <slot />
    </FeatherList>
  </nav>
</template>
<script>
import { FeatherList } from "@featherds/list";

export default {
  props: {
    label: {
      type: String,
      required: true,
    },
  },
  components: {
    FeatherList,
  },
};
</script>
