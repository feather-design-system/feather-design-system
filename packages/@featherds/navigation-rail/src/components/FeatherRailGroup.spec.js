import FeatherRailGroup from "./FeatherRailGroup.vue";
import { shallowMount } from "@vue/test-utils";

const getWrapper = function (options) {
  return shallowMount(FeatherRailGroup, options);
};

describe("FeatherRailGroup.vue", () => {
  it("should render the component open with a label", () => {
    const wrapper = getWrapper({
      propsData: {
        label: "Test",
      },
    });
    expect(wrapper.wrapperElement).toMatchSnapshot();
  });
});
