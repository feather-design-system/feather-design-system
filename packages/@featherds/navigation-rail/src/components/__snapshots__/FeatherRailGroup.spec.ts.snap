// Vitest Snapshot v1

exports[`FeatherRailGroup.vue > should render the component open with a label 1`] = `
<nav
  aria-label="Test"
  class="nav-rail-group"
>
  <feather-list-stub>
    
    
  </feather-list-stub>
</nav>
`;
